recipes/newrelic/apm/dotNet/debian-systemd-nginx.yml

```diff
@@ -0,0 +1,283 @@
+    platform: "ubuntu"
+
+keywords:
+  - Apm
+  - .NET
+  - dotnet
+  - aspnet
+  - core
+
+processMatch: []
+
+validationNrql: "SELECT count(*) from Transaction WHERE host like '{{.HOSTNAME}}%' facet entityGuid since 10 minutes ago"
+
+successLinkConfig:
+   type: EXPLORER
+   filter: '"`tags.language` = ''dotnet''"'
+
+install:
+
+  version: "3"
+  silent: true
+
+  tasks:
+    default:
+      cmds:
+        - task: verify_continue
+        - task: assert_pre_req
+        - task: introspector
+        - task: gather_services
+        - task: stop
+        - task: remove_any_previous
+        - task: install
+        - task: configure
+        - task: start
+        - task: ensure_sites_started
+        - task: cleanup_temp_files
+
+    verify_continue:
+      cmds:
+        - |
+          YELLOW='\033[0;33m'
+          NOCOLOR='\033[0m'
+          echo -e "${YELLOW}
+          ================================================================================
+          =                                                                              =
+          =                                   Warning                                    =
+          =                                                                              =
+          =       This installation will automatically restart all systemd services      =
+          =       that run .NET applications upon completion.                            =
+          =                                                                              =
+          ================================================================================
+          ${NOCOLOR}"
+          echo "
+          If you are hosting your .NET application differently then check out our other installation options:
+          https://docs.newrelic.com/docs/agents/net-agent/installation/install-net-agent-linux/.
+          "
+          NEW_RELIC_ASSUME_YES="{{.NEW_RELIC_ASSUME_YES}}"
+          if [[ "$NEW_RELIC_ASSUME_YES" != "true" ]]; then
+            while :; do
+              echo -n "Do you want to install the .NET Agent Y/N (default: Y)? "
+              read answer
+              echo ""
+              NEW_RELIC_CONTINUE=$(echo "${answer^^}" | cut -c1-1)
+              if [[ -z "$NEW_RELIC_CONTINUE" ]]; then
+                NEW_RELIC_CONTINUE="Y"
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "N" ]]; then
+                echo "Exiting the installation"
+                exit 130
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "Y" ]]; then
+                break
+              fi
+              echo -e "Please type Y or N only."
+            done
+          fi
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_UNZIP_INSTALLED=$(which unzip | wc -l)
+          if [ $IS_UNZIP_INSTALLED -eq 0 ]
+          then
+            echo "unzip is required to run the newrelic install. Please install unzip and re-run the installation." >> /dev/stderr
+            exit 17
```

Comment:
Can we document this specific error code in the docs/error-codes.md

Reply:
Sure.  However, I'm first exploring the possibility of having our LSI made available in `gzip` format so it can be uncompressed using the gzip tool that is more generally available by default on Linux distros.

This is more a question for the owners of this repo, but it seems like if we create a specific error code for every potential tool needed by an install recipe we might run out of unique numbers in the 1-255 range.

Reply:
I think we are going to take a look at this requirement and try to work around it.

Reply:
This has been resolved by hosting a `gzip`ped version of our lsi in S3, so we don't need `unzip` anymore.